src/typed-ast/node.h

```diff
@@ -0,0 +1,52 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+namespace verona::ast
+{
+  class NodePrinter;
+  typedef std::string Symbol;
+
+  /// This represents a location in the source.
+  /// TODO: reuse the code from `compiler/source_manager.h` for a more efficient
+  /// representation.
```

Comment:
It would be nice if this used MLIR-compatible source location, or even reused it to avoid creating infrastructure that exists (and we use).